### PR TITLE
feat: basic ddos protection

### PR DESCRIPTION
**Pull request by nduchak**
This PR introduces basic DDOS protection for file upload.
The flow is next:
 - Increment the counter for each upload from a specific IP address
 - Do not allow to upload files from IP which exceed upload attempts limit per period(configurable)
 - GC mechanism for removing expired entities with upload attempts for IP address
Also, do not log/return stack trace for errors (configurable for each env) and add a check for free disk space on the machine

### CODE REVIEW

File: src/upload/index.ts [hunk 1]
```diff
@@ -46,7 +28,7 @@ const upload: UploadService = {
     // Init upload route
     app.post(
       ServiceAddresses.Upload,
-      uploadMiddleware.array('files'),
+      // uploadMiddleware.array('files'),
```
Comment (artem-brazhnikov) [inline src/upload/index.ts:31]: delete?

File: src/upload/handler/upload.handler.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  return async (req: any, res: any): Promise<void> => {
+    // Check if this IP is not DDOS
+    if (!await isClientAllowedToUpload(req)) {
+      return res.status(400).json({
+        error: 'Not allowed'
+      })
+    }
+
+    uploadMiddleware(req, res, async (err: any) => {
+      if (err) {
```
Comment (artem-brazhnikov) [inline src/upload/handler/upload.handler.ts:71]: I think this callback deserves decomposition. At least 2 parts can be identified: error handling and upload logic.

**Review by artem-brazhnikov — APPROVED** [review]

File: src/upload/handler/upload.handler.ts (hunk 2 above)
Comment (nduchak) [reply src/upload/handler/upload.handler.ts]: done

File: src/upload/index.ts (hunk 1 above)
Comment (nduchak) [reply src/upload/index.ts:31]: done

**Comment by artem-brazhnikov** [comment]
LGTM

**Pull request merged**